packages/core/src/blitz-app-root.tsx

```diff
@@ -110,7 +109,7 @@ export function withBlitzAppRoot(UserAppRoot: React.ComponentType<any>) {
     }, [])
 
     const dehydratedState = props.pageProps.dehydratedState
-      ? SuperJSON.deserialize(props.pageProps.dehydratedState)
+      ? props.pageProps.dehydratedState
```

Comment:
It's interesting to me that removing the deserialisation helps in this case, since babel-plugin-superjson-next should be serializing `dehydratedState`: https://github.com/blitz-js/blitz/blob/a28216d067ff03bbd954a553275972e6f9340bc8/packages/babel-preset/src/index.ts#L16 🤔

Reply:
I pushed an update in 98bdcf8, which should fix the behaviour while retaining SuperJSON support. The problem basically was that `SuperJSON.deserialize` was a totally wrong call - see the 98bdcf8's commit message for more context.

Reply:
Awesome, thanks for making that change!

I think I will add a test for some of the more complex types, like Map!

Reply:
Interestly @Skn0tt, if I swap out `getDate` with `getComplexTypes` and return something like:
```
export default async function getComplexTypes() {
  return {
    date: new Date(0),
    map: new Map().set("foo", "bar"),
    obj: {
      one: 1,
      child: {
        two: 2,
      },
    },
  }
}
```

When `map: new Map()` is in the returned object, the test fails:
```
 NoSuchElementError: no such element: Unable to locate element: {"method":"css selector","selector":"#content"}
      (Session info: headless chrome=92.0.4515.107)

      77 |     it("should work", async () => {
      78 |       const browser = await webdriver(context.appPort, "/dehydrated-state")
    > 79 |       const content = await browser.elementByCss("#content")
```

But when the `map` key is omitted from the return object, the test passes.

Any ideas here, should this be supported?

Reply:
Yes, it should definitely be supported. That's some weird behaviour! Will look into it.

Reply:
Thank you @Skn0tt.
Let me know if I can lend a hand.
I'll park this PR until you've had a look 👍

Reply:
Hey hey, I'm just starting to learn and poke around the blitz codebase, but I tried to debug this issue. I've found that after 98bdcf8, a possibility that the test is failing is that the query keys don't match.

The dehydrated state on `blitz-app-root`is: 
```
{ dehydratedState: 
   { mutations: [],
     queries: 
      [ { state: 
           { data: Wed Dec 31 1969 21:00:00 GMT-0300 (Argentina Standard Time),
             dataUpdateCount: 1,
             dataUpdatedAt: 1628558585243,
             error: null,
             errorUpdateCount: 0,
             errorUpdatedAt: 0,
             fetchFailureCount: 0,
             fetchMeta: null,
             isFetching: false,
             isInvalidated: false,
             isPaused: false,
             status: 'success' },
          queryKey: 
           [ '/api/queries/getDate',
             { json: null, meta: { values: [ 'undefined' ] } } ],
          queryHash: '["/api/queries/getDate",{"json":null,"meta":{"values":["undefined"]}}]' } ] } }
```

while if you log `getQueryKey(getDate, null)` in the client,you get: 
```
{ queryKey: [ '/api/queries/getDate', { json: null } ] }
```


If you look closely you'll see that the latter code is missing the `meta` part of the key: 
```
queryKey: 
           [ '/api/queries/getDate',
             { json: null, meta: { values: [ 'undefined' ] } } ]

vs

{ queryKey: [ '/api/queries/getDate', { json: null } ] }
```

I'm not sure what should be done to fix this, but I hope this helps and also I'm available if you need any helping hand @Skn0tt